src/DualAxisScatterplot.ts

```diff
@@ -0,0 +1,332 @@
+/**
+ * author:  Samuel Gratzl
+ * email:   samuel_gratzl@gmx.at
+ * created: 2016-10-28T11:19:52.797Z
+ */
+
+import {axisLeft, axisBottom, axisRight, AxisScale, Axis} from 'd3-axis';
+import {scaleLinear} from 'd3-scale';
+import {select} from 'd3-selection';
+import {quadtree, Quadtree} from 'd3-quadtree';
+import {ISymbol, ISymbolRenderer, ERenderMode, createRenderer} from './symbol';
+import merge from './merge';
+import {
+  hasOverlap,
+  IBoundsPredicate
+} from './quadtree';
+import {cssprefix, DEBUG, debuglog} from './constants';
+import AScatterplot, {
+  fixScale,
+  IScale,
+  IScatterplotOptions,
+  IScalesObject,
+  IAccessor,
+  EScaleAxes,
+  ERenderReason,
+} from './AScatterplot';
+
+export interface IScalesObjectDualAxis extends IScalesObject {
+  y2: IScale;
+}
+
+/**
+ * scatterplot options
+ */
+export interface IScatterplotOptionsDualAxis<T> extends IScatterplotOptions<T> {
+
+  /**
+   * x2 accessor of the secondary data
+   * default: d.x
+   * @param d
+   */
+  x2?: IAccessor<T>;
+
+  /**
+   * y2 accessor of the secondary data
+   * default: d.y
+   * @param d
+   */
+  y2?: IAccessor<T>;
+
+  /**
+   * y axis label
+   * default: x
+   */
+  y2label?: string;
+
+  /**
+   * d3 y2 scale
+   * default: linear scale with a domain from 0...100
+   */
+  y2scale?: IScale;
+
+  /**
+   * instead of specifying the scale just the y limits
+   */
+  y2lim?: [number, number];
+
+  /**
+   * renderer used to render secondary dataset
+   * default: steelblue circle
+   */
+  symbol2?: ISymbol<T>|string;
+}
+
+//normalized range the quadtree is defined
+const DEFAULT_NORMALIZED_RANGE = [0, 100];
+
+function defaultProps<T>(): IScatterplotOptionsDualAxis<T> {
+  return {
+    x: (d) => (<any>d).x,
+    y: (d) => (<any>d).y,
+
+    x2: (d) => (<any>d).x,
+    y2: (d) => (<any>d).y,
+
+    xlabel: 'x',
+    ylabel: 'y',
+    y2label: 'y2',
+
+    xscale: <IScale>scaleLinear().domain([0, 100]),
+    yscale: <IScale>scaleLinear().domain([0, 100]),
+    y2scale: <IScale>scaleLinear().domain([0, 1000]),
+
+    symbol: 'o',
+    symbol2: 'o',
+  };
+}
+
+/**
+ * a class for rendering a double y-axis scatterplot in a canvas
+ */
+export default class DualAxisScatterplot<T> extends AScatterplot<T> {
+
+  protected readonly normalized2pixel: IScalesObjectDualAxis = {
```

Comment:
as an alternative you could rewrite it to `normalized2pixelX`, `normalized2pixelY`, ...